Update build scripts

KengoTODA opened the pull request:
Hello, thanks for your sharing great idea and implementation. I'm trying to introduce semantic-release to gradle build, and investigating how this plugin is working.

I found several rooms to improve, especially around build script, so I want to suggest:

1. Upgrade Gradle from 2 to 4 (later I will try to use 5), with nebula update that supports Groovy 2.4.10
2. Upgrade FindBugs from 3.0.0 to 3.0.1 ([SpotBugs](https://github.com/spotbugs/spotbugs/) will be next challenge)
3. Remove `buildSrc` that makes build complicated
4. Add missing dependency on `"gradle.plugin.de.gliderpilot.gradle.semantic-release:gradle-semantic-release-plugin:1.4.0"`

Hope that it helps your daily hacking. Thanks in advance :)

KengoTODA committed:
chore: simplify build scripts

KengoTODA committed:
chore: upgrade Gradle from 2 to 3

KengoTODA committed:
chore: upgrade findbugs to 3.0.1

KengoTODA committed:
chore: delete needless file that has been merged to build.gradle

KengoTODA committed:
chore: upgrade Gradle from 3 to 4

KengoTODA committed:
chore: remove needless hack to run test on Gradle 4

tschulte commented:
Thanks for your contribution. I will have a look at it later.

tschulte commented on build.gradle:
```diff
@@ -27,24 +27,24 @@ buildscript {
         classpath 'org.kt3k.gradle.plugin:coveralls-gradle-plugin:2.6.3'
         classpath 'com.github.ben-manes:gradle-versions-plugin:0.13.0'
         classpath "com.gradle.publish:plugin-publish-plugin:0.9.9"
+        classpath "gradle.plugin.de.gliderpilot.gradle.semantic-release:gradle-semantic-release-plugin:1.4.0"
```
I think, this should be 1.+. Otherwise we would have to upgrade the version manually.

```suggestion
        classpath "gradle.plugin.de.gliderpilot.gradle.semantic-release:gradle-semantic-release-plugin:1.+"
```

tschulte reviewed (changes requested):
The whole point of the buildSrc has been to eat our own dogfood. When the plugin is broken, it will not release. But there are two catches:

1. While refactoring, the build might not work. This might even result in the IDE to have problems
2. The build is slightly more complicated

That being said, I think you are right in fixing this. I have only one change request regarding using a version range for the dependency on ourself.

tschulte committed:
Update build.gradle

Keep using latest version of gradle-semantic-release-plugin for better dog-fooding.

Co-Authored-By: KengoTODA <skypencil+github@gmail.com>

KengoTODA commented:
Thanks for your feedback! The usage shared by you is new for me, I've learned much from that.

I've fixed version range, please have a review. I'm not sure where is the conflict, master branch has no update recently... could you check it?

tschulte commented:
I did merge branch develop last week and removed that branch. There was a conflicting file change in buildSrc/build.gradle. I fixed that.

KengoTODA commented:
Thank you!
